tcs-persistence/src/main/resources/queries/traineeInfoForConnection.sql

```diff
@@ -62,6 +62,12 @@ from (
     ) currentGrade on cd.id = currentGrade.traineeId
   WHERECLAUSE(cd, id)
 ) as ot
+-- Filter based on "role" from the "Person" table
+WHERE ot.personId NOT IN (
+  SELECT DISTINCT p.id
+  FROM Person p
+  WHERE LOWER(p.role) LIKE '%dummy%' OR LOWER(p.role) LIKE '%placeholder%'
+)
```

Comment:
Can you put the sql operations into lowercase to match the rest of the file

This does appear to make the query perform slower, don't have any changes to suggest at this stage though

Reply:
We can try this on Stage. 

If there's a timeout or performance issue, we can get all the non-dummy records via jpa query, and then have a filter in Java.